core/src/main/java/org/jdbi/v3/BuiltInArgumentFactory.java

```diff
@@ -36,14 +41,14 @@
  * class directly, but instead should bind your object with the
  * {@link SQLStatement} convenience methods.
  */
-public class BuiltInArgumentFactory<T> implements ArgumentFactory<T> {
+public class BuiltInArgumentFactory implements ArgumentFactory<Object> {
```

Comment:
Yeah, this generics stuff is tricky.  I'm still not sure how we decide which things have a `<T>` or not in a principled manner.  I'm +1 to killing this one, though.